packages/sanity-plugin-nacelle-input/README.md

```diff
@@ -0,0 +1,147 @@
+### Install Peer Dependencies
+
+`npm i @sanity/ui styled-components`
+
+### Install the plugin
+
+`sanity install @nacelle/sanity-plugin-nacelle-input`
+
+### Credentials
+
+You'll need to provide the ID and Token associated with your Nacelle space. These credentials can be found in the [Nacelle Dashboard](https://dashboard.getnacelle.com/).
+
+#### Adding credentials to a single space:
+
+##### Config file for single space - `./config/@nacelle/sanity-plugin-nacelle-input.json`
+
+```json
+{
+  "nacelleSpaceEndpoint": "your-nacelle-storefront-api-endpoint",
+  "nacelleSpaceToken": "your-nacelle-graphql-token"
+}
+```
+
+##### in `.env.development` / `.env.production`
+
+```
+SANITY_STUDIO_NACELLE_SPACE_ENDPOINT=your-nacelle-storefront-api-endpoint
+SANITY_STUDIO_NACELLE_SPACE_TOKEN=your-nacelle-graphql-token
+```
+
+
+#### Adding Credentials to multiple spaces
+
+Create a config file at `./config/@nacelle/sanity-plugin-nacelle-input.json`. Then add a `nacelleSpaces` array like so:
+
+```json
+"nacelleSpaces": [
+    {
+      "spaceName": "Space 1",
+      "nacelleSpaceEndpoint": "https://your-nacelle-storefront-api-endpoint",
+      "nacelleSpaceToken": "your-nacelle-graphql-token"
+    },
+    {
+      "spaceName": "Space 2",
+      "nacelleSpaceEndpoint": "https://storefront.api.development.nacelle.com/graphql/v1/spaces/clever-owl-jr0WwlZv7L",
+      "nacelleSpaceToken": "2a74743f-7a00-4274-9cb6-2dfe15e89d47"
+    }
+  ]
+```
+
+## Use in Schema Documents
+
+Set the `type` field to `nacelleData` to use the custom input component:
+
+```js
+{
+  name: 'handle',
+  title: 'Handle',
+  type: 'nacelleData',
+}
+```
+
+### Options
+
+By default, the custom input component allows you to choose a `handle` from either products or collections.
+
+Realistically, you probably want to restrict the component to _either_ products or collections. To do that, provide either `['products']` or `['collections']` to `options.dataType`:
+
+```js
+// example: collections ONLY
+{
+  name: 'collectionHandle',
+  title: 'Collection',
+  type: 'nacelleData',
+  options: {
+    dataType: ['collections']
+  }
+}
+```
+
+```js
+// example: products ONLY
+{
+  name: 'productHandle',
+  title: 'Product',
+  type: 'nacelleData',
+  options: {
+    dataType: ['products']
+  }
+}
+```
+
+### Using This Data in Your Frontend Project
+
+Since this custom input component just stores the `handle` of a particular product or collection, you'll use the [Nacelle Client JS SDK](https://docs.getnacelle.com/api-reference/client-js-sdk.html) to fetch the associated product or collection object.
+
+#### Product
+
+```js
+const product = await client.data.product({
+  handle: 'handle-from-my-sanity-entry'
+})
+```
+
+#### Collection
+
+```js
+const collection = await client.data.collection({
+  handle: 'handle-from-my-sanity-entry'
+})
+
+const productHandles = collection.productLists[0].handles
+
+const collectionProducts = await client.data.products({
+  handles: productHandles
+})
+```
```

Comment:
This needs to be updated to reflect Nacelle v2 conventions. We can link to [`@nacelle/storefront-sdk`](https://nacelle.com/docs/querying-data/storefront-sdk) docs and [Storefront API](https://nacelle.com/docs/querying-data/storefront-api) docs.

Reply:
Should be all set now